pages/Posts.tsx

```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect, useRef } from "react";
+
+const BASE_URL = "https://jsonplaceholder.typicode.com/posts?_page=";
+
+export default function Posts() {
+  type Post = {
+    id: number;
+    title: string;
+    body: string;
+    userId: number;
+  };
+
+  type Error = {
+    message: string;
+  };
+  type Page = {
+    page: number;
+  };
+
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [page, setPage] = useState<Page>({ page: 1 });
+  const hasMore = page.page * 1 < 100;
+
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+
+  const fetchPosts = async () => {
+    try {
+      setLoading(true);
+      const response = await fetch(`${BASE_URL}${page.page}`);
+      if (!response.ok) {
+        throw new Error("Failed to fetch posts");
+      }
+      const posts = await response.json();
+      setPosts((prev) => [...prev, ...posts]);
+    } catch (error: any) {
+      setError(error.message);
+    } finally {
+      setLoading(false);
+    }
+  };
+  useEffect(() => {
+    setTimeout(() => {
+      fetchPosts();
+    }, 1000);
+  }, [page]);
```

Comment:
Why?

Reply:
To schedule the post to render at 1 second